Issue of MYSQL_OPT_SSL_MODE

Currently there are many `#ifdef MYSQL_OPT_SSL_MODE` in `drv_mysql.c`.
https://github.com/akopytov/sysbench/blob/master/src/drivers/mysql/drv_mysql.c#L67

But where does this macro `MYSQL_OPT_SSL_MODE` be defined? In `mysql.h`, it is only a value of enum.
I don't think all compilers can recognize this.

In my test environment:
```
Ubuntu 18.04.1 LTS
gcc (Ubuntu 7.3.0-27ubuntu1~18.04) 7.3.0
libmysqlclient-dev/bionic-updates,bionic-security,now 5.7.24-0ubuntu0.18.04.1 amd64 [installed]
```
The compilation won't go into the #ifdef blocks, and the sysbench built has SSL issues.
Also, the output of `sysbench --help` contains the wrong parameter:
`--mysql-ssl[=on|off]             use SSL connections, if available in the client library [off]`

Besides, the code inside `#ifdef` also has issue.
After removed the `#ifdef` blocks, the compilation aborted with error `'ssl_mode_typelib' undeclared`.
```
drv_mysql.c: In function ‘mysql_drv_init’:
drv_mysql.c:269:46: error: ‘ssl_mode_typelib’ undeclared (first use in this function); did you mean ‘ssl_mode_string’?
   args.ssl_mode = find_type(ssl_mode_string, ssl_mode_typelib, FIND_TYPE_BASIC);
                                              ^~~~~~~~~~~~~~~~
                                              ssl_mode_string
```

--- a/m4/sb_check_mysql.m4
+++ b/m4/sb_check_mysql.m4
@@ -114,7 +114,24 @@ AC_DEFINE([USE_MYSQL], 1,
 USE_MYSQL=1
 AC_SUBST([MYSQL_LIBS])
 AC_SUBST([MYSQL_CFLAGS])
+
+AC_MSG_CHECKING([if mysql.h defines MYSQL_OPT_SSL_MODE])
+
+SAVE_CFLAGS="${CFLAGS}"
+CFLAGS="${CFLAGS} ${MYSQL_CFLAGS}"
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM(
+        [[
+#include <mysql.h>
+
+enum mysql_option opt = MYSQL_OPT_SSL_MODE;
+        ]])], [
+          AC_DEFINE([HAVE_MYSQL_OPT_SSL_MODE], 1,
+                    [Define to 1 if mysql.h defines MYSQL_OPT_SSL_MODE])
+          AC_MSG_RESULT([yes])
+          ], [AC_MSG_RESULT([no])])
 ])
+CFLAGS="${SAVE_CFLAGS}"
+
 
 AM_CONDITIONAL([USE_MYSQL], test "x$with_mysql" != xno)
 AC_SUBST([USE_MYSQL])

--- a/src/drivers/mysql/drv_mysql.c
+++ b/src/drivers/mysql/drv_mysql.c
@@ -323,10 +323,10 @@ static int mysql_drv_real_connect(db_mysql_conn_t *db_mysql_con)
 
     mysql_ssl_set(con, ssl_key, ssl_cert, ssl_ca, NULL, args.ssl_cipher);
 
-#ifdef MYSQL_OPT_SSL_MODE
+#ifdef HAVE_MYSQL_OPT_SSL_MODE
     unsigned int opt_ssl_mode = SSL_MODE_REQUIRED;
 
-    DEBUG("mysql_options(%p,%s,%u)", con, "MYSQL_OPT_SSL_MODE", opt_ssl_mode);
+    DEBUG("mysql_options(%p, %s, %u)", con, "MYSQL_OPT_SSL_MODE", opt_ssl_mode);
     mysql_options(con, MYSQL_OPT_SSL_MODE, &opt_ssl_mode);
 #endif
   }

